src/shared/misc.c

```diff
@@ -521,6 +521,44 @@ pid_t get_pid(const char *applet,const char *pidfile)
 	return pid;
 }
 
+struct ready ready_parse(const char *applet, const char *ready_string)
+{
+	struct ready ready = {0};
+	if (sscanf(ready_string, "fd:%d", &ready.fd) != 1)
+		eerrorx("%s: invalid ready '%s'.", applet, optarg);
+
+	ready.type = READY_FD;
+	pipe(ready.pipe);
```

Comment:
Return value needs checking here, it may fail.

Same with some of the `dup2` calls.